### PR TITLE
 nixos/no-x-libs: add gst-plugins-bad, gst-plugins-rs

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
The gst-plugins-base hack is very similar to https://github.com/NixOS/nixpkgs/pull/231185/files#diff-bd924545a322bae41a5f57f50113281417c8ea50fec7801f606840410f492164R169

File: pkgs/development/libraries/gstreamer/rs/default.nix [hunk 1]
```diff
@@ -137,6 +138,12 @@ stdenv.mkDerivation rec {
     '';
   };
 
+  # gstreamer-gl-* detection is broken due to an upstream meson issue
+  postPatch = lib.optionalString (!gst-plugins-base.glEnabled) ''
+    substituteInPlace meson.build \
+      --replace "['gstreamer-gl-1.0', 'gst-plugins-base', 'gst_gl_dep', 'gstgl']," ""
+  '';
```
Comment (lilyinstarlight) [inline pkgs/development/libraries/gstreamer/rs/default.nix:145]: Look at the `preConfigure` hook which already does this on non-aarch64-darwin. It needs to be conditionalized on `gst-plugins-base.glEnabled` instead and then this should be automatic

File: pkgs/development/libraries/gstreamer/rs/default.nix [hunk 2]
```diff
@@ -101,7 +102,7 @@ let
       "reqwest" # tests hang on darwin
       "threadshare" # tests cannot bind to localhost on darwin
       "webp" # not supported on darwin (upstream crate issue)
-    ] ++ lib.optionals (stdenv.isDarwin && !stdenv.isAarch64) [
+    ] ++ lib.optionals ((stdenv.isDarwin && !stdenv.isAarch64) || !gtkSupport) [
```
Comment (lilyinstarlight) [inline pkgs/development/libraries/gstreamer/rs/default.nix:105]: Can you just make this `!gtkSupport` and then adjust the default value for `gtkSupport` to be `gst-plugins-base.glEnabled`?

File: pkgs/development/libraries/gstreamer/bad/default.nix [hunk 3]
```diff
@@ -297,10 +299,11 @@ stdenv.mkDerivation rec {
     "-Ddoc=disabled" # needs gstcuda to be enabled which is Linux-only
     "-Dnvcodec=disabled" # Linux-only
     "-Dva=disabled" # see comment on `libva` in `buildInputs`
+  ] ++ lib.optionals (stdenv.isDarwin || !guiSupport) [
+    "-Ddirectfb=disabled"
```
Comment (lilyinstarlight) [inline pkgs/development/libraries/gstreamer/bad/default.nix:303]:
```suggestion
  ] ++ lib.optionals (!stdenv.isLinux || !guiSupport) [
    "-Ddirectfb=disabled"
```

Should be conditional on `stdenv.isLinux` to match the earlier `buildInput` conditional

**Review by lilyinstarlight — COMMENTED** [review]
Thank you for this! Left a few comments and will test this with that other PR later

File: pkgs/development/libraries/gstreamer/rs/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/gstreamer/rs/default.nix:145]: I can't find the upstream issue that mentioned that anymore

**Review by lilyinstarlight — APPROVED** [review]
Looks great now, thank you!

I tested build of gst-plugins-bad and gst-plugins-rs with and without xlibs successfully on x86_64-linux

**Pull request merged**